discovery_child_development/pipeline/models/binary_classifier/04b_train_distilbert_classifier.py

```diff
@@ -0,0 +1,203 @@
+
+Optional arguments:
+--production : Determines whether to create the embeddings for the full dataset or a test sample (default: True)
+--wandb : Determines whether a run gets logged on wandb (default: False)
+
+"""
+import wandb
+import numpy as np
+import argparse
+from nesta_ds_utils.loading_saving import S3
+from discovery_child_development.getters.binary_classifier.embeddings_hugging_face import (
+    get_embeddings,
+)
+from discovery_child_development.utils.huggingface_pipeline import (
+    load_model,
+    load_training_args,
+    load_trainer,
+    saving_huggingface_model,
+)
+from discovery_child_development.utils.general_utils import replace_binary_labels
+from discovery_child_development.utils import wandb as wb
+from discovery_child_development.utils import classification_utils
+from discovery_child_development.getters.binary_classifier.gpt_labelled_datasets import (
+    get_labelled_data_for_classifier,
+)
+from discovery_child_development.utils.testing_examples_utils import (
+    testing_examples_huggingface,
+)
+from discovery_child_development import (
+    logging,
+    S3_BUCKET,
+    config,
+    binary_config,
+    PROJECT_DIR,
+)
+from transformers import set_seed
+
+# Set up
+S3_PATH = "models/binary_classifier/"
+VECTORS_PATH = "data/labels/binary_classifier/vectors/"
+VECTORS_FILE = "distilbert_sentence_vectors_384_labelled"
+SEED = config["seed"]
+NUM_SAMPLES = config["labelled_sample_size"]
+# Set the seed
+set_seed(SEED)
+
+if __name__ == "__main__":
+    # Set up the command line arguments
+    parser = argparse.ArgumentParser(
+        description="Script to run with command line arguments."
+    )
+
+    parser.add_argument(
+        "--production",
+        type=bool,
+        default=True,
+        help="Do you want to run the code in production? (default: True)",
+    )
+
+    parser.add_argument(
+        "--wandb",
+        type=bool,
+        default=False,
+        help="Do you want to log this as a run on wandb? (default: False)",
+    )
+    # Parse the arguments
+    args = parser.parse_args()
+    logging.info(args)
+
+    # Loading the training and validation embeddings
+    embeddings_training = get_embeddings(
+        identifier="",
+        production=args.production,
+        set_type="train",
+        vectors_path=VECTORS_PATH,
+        vectors_file=VECTORS_FILE,
+    )
+    embeddings_validation = get_embeddings(
+        identifier="",
+        production=args.production,
+        set_type="validation",
+        vectors_path=VECTORS_PATH,
+        vectors_file=VECTORS_FILE,
+    )
+
+    if args.wandb:
+        logging.info("Logging run on wandb")
+        run = wandb.init(
+            reinit=True,
+            project="ISS supervised ML",
+            job_type="Binary classifier - huggingface",
+            save_code=True,
+            tags=["huggingface", "binary_classifier", "sentence_embeddings", "all"],
```

Comment:
This is a really minor thing but seeing as the job type is "binary classifier", I don't think you need this in the tags as well?

Also what does the "sentence_embeddings" tag mean? The simple classifiers are trained on sentence embeddings from the all-miniLM model, whereas this code uses embeddings created with distilbert, right? Or is that actually wrong?

Reply:
No clue really! Will remove haha, I might just specify where it is miniLM vs distilbert